Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,131 @@
+
+This rule would be rather simple to enforce via Danger, using the following regex: `class .*ViewModel\W` inside every test file. 
+To do so, we should whitelist every file that matched `Tests.*`.
+
+Nevertheless, as of June 7th, 2019, we have 254 occurences of this rule in 251 files. 
+
+As such, I believe this rule should only be enforced via Danger because using SwiftLint to monitor this would trigger 254 new warnings  (obviously). Since we already have a tremendous amount of warnings (~500), we'd have to refactor all of these new warnigs just to get back to our starting point. This would require a rather significant effort for very little benefit.
+
+On the other hand, enforcing the rule with Danger wouldn't cause those issues but we would only be able to detect new occurences of this anti-pattern since Danger will only inspect changes made in pull requests.
+
+### Monitoring ViewController lifecycle directly instead of relying on `ScreenLifecycleEvent`
+
+With the introduction of `ScreenLifecycleEvent`, developers should no longer manually monitor the viewController's lifecycle events.
+Instead, this kind of monitoring can now easily be done through `BoxViewModel`'s that conform to `ScreenLifecycleAware` via `send(_: ScreenLifecycleEvent)`.
+
+Here's a brief example of how it was typically done in both Builders and FlowControllers:
+
+```swift
+viewController
+            .reactive
+            .signal(for: #selector(UIViewController.viewWillDisappear))
+            .discardValues()
+            .observeValues(dismiss)
+```
+
+Nowadays, this can be done much more elegantly (thanks Anders!):
+
+```swift
+func send(_ event: ScreenLifecycleEvent) {
+        switch (event, state.status) {
+        case (.didAppear, .showing):
+            runSomething()
+        default:
+            break
+        }
+    }
+```
+
+As of June 7th, 2019, we still have 16 results in 13 different files.
+
+In order to detect this, we could simply monitor occurences of the following regex:
+
+`\.signal\(for: \#selector\(UIViewController\.`
+
+We'd have to restrict this rule to Builders and FlowControllers only, therefore we'd have to whitelist filenames that matched `.*Builder.swift` and `.*FlowController.swift`.
+
+This rule could be enforced with either tool but I'd recommend using SwiftLint for this since due to the low number of warnings it would trigger in the current state of our database. Also, I believe we should try to use Danger for particularly important warnings in order not to spam the pull request with comments.
+
+
+### Deprecated calls of DesignLibrary components (aka pre-NVL components)
+
+Given that NVL is scheduled for release within two weeks, using NVL components is pivotal to Babylon's overall look and feel.
+With the exception of the empty space component (even though it has its own NVL equivalent), every other pre-NVL component should trigger a warning.
```

Comment:
I think we were using this because there was no NVL equivalent - if we have it now let's just switch to it and do not make any exceptions.